src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
I think the main reason that we need a separate tsi registration mechanism is that we do not want grpc (grpc_init or grpc_shutdown) directly depend on gts or other tsi implementations, but rather let grpc depend on tsi and let tsi depend on gts and other implementations, which represents a good layering model. How do you think? 

Regarding the new interface you proposed, the main concern I have is on init(). I assume the init should be invoked within grpc_init() that will be called before any operation including the invocation of tsi_register_implementation. Isn't it contradictory? For destroy(), we do not have this problem since we just need to register the destroy() before grpc_shutdown() is invoked, which could be anywhere in tsi implementation. How do you think?